### PR TITLE
chore: add transaction bench and web bench

### PR DESCRIPTION
**Pull request by SimonThormeyer**
# What's new in this PR
Adds benches and runs them in CI. The benches were used for an investigation regarding transactions and group store.

----
##### PR Submission Checklist for internal contributors

- The **PR Title**
  - [ ] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
  - [ ] contains a reference JIRA issue number like `SQPIT-764`
  - [ ] answers the question: _If merged, this PR will: ..._ ³

1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.

### CODE REVIEW

File: crypto/benches/transaction.rs [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+pub mod utils;
+
+use utils::*;
+fn decrypt_transaction(c: &mut Criterion) {
+    const MESSAGE_LENGTH: usize = 100;
+    const MESSAGE_COUNT: usize = 1000;
+
+    let mut group = c.benchmark_group(format!(
+        "Decrypting {MESSAGE_COUNT} messages, transaction vs. no transaction"
+    ));
```
Comment (coriolinus) [inline crypto/benches/transaction.rs:15]: This group name shows up in the output, and it's not obvious which case is running for any given results. Recommend creating the group slightly later within the `transactional` loop, and naming it like `Decrypting {MESSAGE_COUT} messages: {id}`.

File: crypto/benches/transaction.rs [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+        });
+    }
+    group.finish();
+}
+
+criterion_group!(
+    name = decryption;
+    config = criterion();
+    targets =
+    decrypt_transaction,
```
Comment (coriolinus) [inline crypto/benches/transaction.rs:72]: This would be easier to read if it were on a single line.

**Review by coriolinus — APPROVED** [review]
I suspect the CI failures are spurious, but it would be good after https://github.com/wireapp/core-crypto/pull/789#pullrequestreview-2469742941 is merged to rebase this atop that to ensure that it gets clean runs.

Approving anyway for now; the code looked right, except for some nits below.

File: crypto/benches/transaction.rs (hunk 1 above)
Comment (SimonThormeyer) [reply crypto/benches/transaction.rs:15]: @coriolinus the output looks like this:
`Decrypting 1000 messages, transaction vs. no transaction/transactional` and
`Decrypting 1000 messages, transaction vs. no transaction/non-transactional`, where the first is transactional sending and the second non-transactional.
I'd like to keep the group creation in that place, as I follow the pattern of the other existing benches. I admint those might need an overhaul, though.

However, I made the output a bit more concise and added the information of message length in bytes.

Comment (coriolinus) [reply crypto/benches/transaction.rs]: Yes, that's clearer now, thanks!

**Pull request merged**